TestLowSequenceHandlingWithAccessGrant fails against integration test

`TestLowSequenceHandlingWithAccessGrant ` is not passing against Couchbase Server in non-XATTR mode.  It fails w/ this error:

```
2017-11-15 14:08:39.881244 I | Changes entry: {Seq:2::5, ID:doc-5, Changes:[map[rev:1-a]]}
2017-11-15 14:08:39.881254 I | Changes entry: {Seq:2::6, ID:doc-6, Changes:[map[rev:1-a]]}
2017-11-15 14:08:39.881263 I | Changes entry: {Seq:2::9, ID:doc-9, Changes:[map[rev:1-a]]}
2017-11-15 14:08:49.881408 I | Miscount, count (3) != numEntries (4)

	change_cache_test.go:681
	assert.True(t, err == nil)
2017-11-15 14:08:49.881637 I | Got changes: [{Seq:1, ID:doc-1, Changes:[map[rev:1-a]]} {Seq:2, ID:doc-2, Changes:[map[rev:1-a]]} {Seq:2::6, ID:doc-6, Changes:[map[rev:1-a]]} {Seq:2::5, ID:doc-5, Changes:[map[rev:1-a]]} {Seq:2::6, ID:doc-6, Changes:[map[rev:1-a]]} {Seq:2::9, ID:doc-9, Changes:[map[rev:1-a]]}]

	change_cache_test.go:686
	assert.Equals(t, len(changes), expectedNumChanges)

	expected: 7
	     got: 6
2017-11-15 14:08:49.881742 I | verifyChangesFullSequences: changes size (6) not equals to sequences size (7)

	change_cache_test.go:687
	assert.True(t, verifyChangesFullSequences(changes, []string{"1", "2", "2::6", "2:8:5", "2:8:6", "2::8", "2::9"}))
2017-11-15T14:08:49.881-08:00 WARNING: Error closing GoCB bucket: connection shut down. -- base.CouchbaseBucketGoCB.Close() at bucket_gocb.go:2065
--- FAIL: TestLowSequenceHandlingWithAccessGrant (21.55s)
	assert.go:31:
	assert.go:31:
	assert.go:31:
2017-11-15 14:08:49.881839 I | bucketDataSource.Close() called
FAIL
```

--- a/db/change_cache_test.go
+++ b/db/change_cache_test.go
@@ -610,13 +610,9 @@ func TestLowSequenceHandlingAcrossChannels(t *testing.T) {
 // user gets added to a new channel with existing entries (and existing backfill)
 func TestLowSequenceHandlingWithAccessGrant(t *testing.T) {
 
-	// TODO: this should be fixed for the Non-XATTR case and re-enabled
-	// TODO: details here: https://github.com/couchbase/sync_gateway/issues/3075
-
-	if !base.UnitTestUrlIsWalrus() {
-		t.Skip("This test is only working against Walrus currently.  Needs more investigation.  Failure logs: https://gist.github.com/tleyden/7011c68aa85dd739babf90a1a556469d")
+	if base.TestUseXattrs() {
+		t.Skip("This test does not work with XATTRs due to calling WriteDirect().  Skipping.")
 	}
-
 
 	var logKeys = map[string]bool{
 		"Sequence": true,
@@ -665,7 +661,9 @@ func TestLowSequenceHandlingWithAccessGrant(t *testing.T) {
 	assert.Equals(t, len(changes), 3)
 	assert.True(t, verifyChangesFullSequences(changes, []string{"1", "2", "2::6"}))
 
-	db.Bucket.Incr("_sync:seq", 7, 0, 0)
+	_, incrErr := db.Bucket.Incr("_sync:seq", 7, 7, 0)
+	assert.True(t, incrErr == nil)
+
 	// Modify user to have access to both channels (sequence 2):
 	userInfo, err := db.GetPrincipal("naomi", true)
 	assert.True(t, userInfo != nil)

